pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java

```diff
@@ -49,13 +49,14 @@ public RealtimeSegmentStatsHistory getStatsHistory() {
     return _statsHistory;
   }
 
-  protected void initMemoryManager(RealtimeTableDataManager realtimeTableDataManager, boolean isOffHeapAllocation, String segmentName) {
+  protected void initMemoryManager(RealtimeTableDataManager realtimeTableDataManager, boolean isOffHeapAllocation,
+      boolean isDirectAllocation, String segmentName) {
     ServerMetrics serverMetrics = realtimeTableDataManager.getServerMetrics();
-    if (isOffHeapAllocation) {
+    if (isOffHeapAllocation && !isDirectAllocation) {
```

Comment:
We should just rename the 'isOffHeapAllocation' to 'isMmapAllocation' because both of them are off-heap.

Reply:
i will add a comment here. The variables are derived from the configs. If offheap is not enabled, then we still need a mem manager for fwd index, but dictionaries are on-heap.